### ISSUE TITLE
Prevent events loss on app force close

### ISSUE THREAD

**Issue by azarovalex** [mapbox/mapbox-navigation-ios#3173]
When a user force closes the app, we receive the `applicationWillTerminate` notification. At this moment we need to send all queued events to the server. The SDK dispatches this job on a background thread asynchronously, iOS see that the main thread is free and kills the app before the background job is ready. I conducted a few simple experiments to verify this behavior.

**Doesn't print anything:**
```
func applicationWillTerminate(_ notification: NSNotification) {
    DispatchQueue.global().async {
        Thread.sleep(forTimeInterval: 2)
        print("background job is done")
    }
}
```

**Prints the message:**
```
func applicationWillTerminate(_ notification: NSNotification) {
    DispatchQueue.global().async {
        Thread.sleep(forTimeInterval: 2)
        print("background job is done")
    }
    Thread.sleep(forTimeInterval: 3)
}
```

**TL;DR:** we need to block the main thread for a couple of seconds to make sure the network request will fire. According to the iOS documentation, we will have ~5 seconds, so let's block the thread for this duration.
